src/background/http-instrument.ts

```diff
@@ -231,12 +232,22 @@ export class HttpInstrument {
     );
     */
 
+    const tab =
+      details.tabId > -1
+        ? await browser.tabs.get(details.tabId)
+        : { windowId: undefined, incognito: undefined, url: undefined };
```

Comment:
Indeed an artifact of using TypeScript, and using `undefined` here was a lazy choice to not have to wrap all tab-relevant assignments in if-clauses (so that they are truly undefined if the tab object is not available). Added a code cleanup issue to catch this technical debt: https://github.com/mozilla/openwpm-webext-instrumentation/issues/21